src/core/store.ts

```diff
@@ -40,10 +56,52 @@ const reducer = {
   views: viewsSlice.reducer,
 };
 
+const listenerMiddleware = createListenerMiddleware();
+
+listenerMiddleware.startListening({
+  actionCreator: campaignDeleted,
+  effect: (action) => {
+    const orgId = action.payload[0];
+    Router.push(`/organize/${orgId}/projects`);
+  },
+});
+
+listenerMiddleware.startListening({
+  actionCreator: campaignCreated,
+  effect: (action) => {
+    const campaign = action.payload;
+    Router.push(
+      `/organize/${campaign.organization?.id}/projects/${campaign.id}`
+    );
+  },
+});
+
+listenerMiddleware.startListening({
+  actionCreator: callAssignmentCreated,
+  effect: (action) => {
+    const [callAssignment, campId] = action.payload;
+    Router.push(
+      `/organize/${callAssignment.organization?.id}/projects/${campId}/callassignments/${callAssignment.id}`
+    );
+  },
+});
+
+listenerMiddleware.startListening({
+  actionCreator: surveyCreated,
+  effect: (action) => {
+    const [survey, campId] = action.payload;
+    Router.push(
+      `/organize/${survey.organization.id}/projects/${campId}/surveys/${survey.id}`
+    );
+  },
+});
+
```

Comment:
I think this is a great experiment and we should definitely keep it for now. But I'm wondering if this actually is unnecessary, because it could (in theory) be replaced with simple `await`s in the components. Will try to find an example and propose it there (but again, lets keep this for now and reflect on what is the right path forward).